go/apps/ctrl/services/build/backend/depot/create_build.go

```diff
@@ -0,0 +1,350 @@
+		return nil, connect.NewError(connect.CodeInternal,
+			fmt.Errorf("failed to create build: %w", err))
+	}
+
+	s.logger.Info("Depot build created",
+		"build_id", buildResp.ID,
+		"depot_project_id", depotProjectID,
+		"unkey_project_id", req.Msg.UnkeyProjectId)
+
+	var buildErr error
+	defer buildResp.Finish(buildErr)
+
+	s.logger.Info("Acquiring build machine",
+		"build_id", buildResp.ID,
+		"platform", "arm64",
+		"unkey_project_id", req.Msg.UnkeyProjectId)
+
+	buildkit, buildErr := machine.Acquire(ctx, buildResp.ID, buildResp.Token, "arm64")
+	if buildErr != nil {
+		s.logger.Error("Acquiring depot build failed",
+			"error", buildErr,
+			"build_id", buildResp.ID,
+			"depot_project_id", depotProjectID,
+			"unkey_project_id", req.Msg.UnkeyProjectId)
+		return nil, connect.NewError(connect.CodeInternal,
+			fmt.Errorf("failed to acquire machine: %w", buildErr))
+	}
+	defer buildkit.Release()
+
+	s.logger.Info("Build machine acquired, connecting to buildkit",
+		"build_id", buildResp.ID,
+		"unkey_project_id", req.Msg.UnkeyProjectId)
+
+	buildkitClient, buildErr := buildkit.Connect(ctx)
+	if buildErr != nil {
+		s.logger.Error("Connection to depot build failed",
+			"error", buildErr,
+			"build_id", buildResp.ID,
+			"depot_project_id", depotProjectID,
+			"unkey_project_id", req.Msg.UnkeyProjectId)
+		return nil, connect.NewError(connect.CodeInternal,
+			fmt.Errorf("failed to connect to buildkit: %w", buildErr))
+	}
+
+	// INFO: "s.registryUrl", "depotProjectID" order of these two arg must never change, otherwise depot will decline the registry upload.
+	imageName := fmt.Sprintf("%s/%s:%s-%s", s.registryUrl, depotProjectID, req.Msg.UnkeyProjectId, req.Msg.DeploymentId)
+
+	dockerfilePath := req.Msg.GetDockerfilePath()
+	if dockerfilePath == "" {
+		dockerfilePath = "Dockerfile"
+	}
+
+	s.logger.Info("Starting build execution",
+		"image_name", imageName,
+		"dockerfile", dockerfilePath,
+		"platform", "linux/arm64",
+		"build_id", buildResp.ID,
+		"unkey_project_id", req.Msg.UnkeyProjectId)
+
+	solverOptions := client.SolveOpt{
+		Frontend: "dockerfile.v0",
+		FrontendAttrs: map[string]string{
+			"platform": "linux/arm64",
+			"context":  contextURL,
+			"filename": dockerfilePath,
+		},
+		Session: []session.Attachable{
+			authprovider.NewDockerAuthProvider(authprovider.DockerAuthProviderConfig{
+				ConfigFile: &configfile.ConfigFile{
+					AuthConfigs: map[string]types.AuthConfig{
+						s.registryUrl: {
+							Username: s.username,
+							Password: s.accessToken,
+						},
+					},
+				},
+			}),
+		},
+		Exports: []client.ExportEntry{
+			{
+				Type: "image",
+				Attrs: map[string]string{
+					"name":           imageName,
+					"oci-mediatypes": "true",
+					"push":           "true",
+				},
+			},
+		},
+	}
+
+	buildStatusCh := make(chan *client.SolveStatus, 10)
+	go func() {
+		enc := json.NewEncoder(os.Stdout)
+		enc.SetIndent("", "  ")
+		for status := range buildStatusCh {
+			_ = enc.Encode(status)
+		}
+	}()
```

Comment:
I don't think we need to spam stdout will the whole output wdyt?

Maybe it'd make sense to buffer it and only print it when something goes wrong?

Reply:
You are right. This was kinda helpful when testing locally to see the big picture, but for now errors are enough for us. Do you have an example of that buffer usage?